eng/pipelines/templates/steps/archetype-sdk-docs.yml

```diff
@@ -24,128 +24,18 @@ steps:
       -OutFile "docfx.zip" | Wait-Process; Expand-Archive -Path "docfx.zip" -DestinationPath "./docfx/"
 
       Write-Host "Restore eng/docgeneration/assets/docgen.csproj, to get ECMA2Yml and popimport"
-      dotnet restore "$(Build.SourcesDirectory)/eng/docgeneration/assets/docgen.csproj" /p:BuildBinariesDirectory=$(Build.BinariesDirectory)
+      dotnet restore "$(Build.SourcesDirectory)/doc/ApiDocGeneration/assets/docgen.csproj" /p:BuildBinariesDirectory=$(Build.BinariesDirectory)
```

Comment:
`"$(Build.SourcesDirectory)/doc/ApiDocGeneration` this path is in lots of places which means it has to stay in sync so we should capture it in a variable and pass it through.

Reply:
I still feel like this path should be captured into a variable and passed through. Right now multiple places in the ps1 and yml files have this duplicated and if we ever need to change it we will have to try and find all those locations and update them consistently.

Reply:
Fixed by passing it as a parameter in `eng/pipelines/templates/jobs/archetype-sdk-client.yml` where the `archetype-sdk-doc.yml` template is called.